design/authenticators-status/authn-status-new-implem.md

```diff
@@ -0,0 +1,109 @@
+For each authenticator, a suite of [general](https://github.com/cyberark/conjur/blob/master/design/authenticators-status/authn_status_general.md) 
+checks are run followed by those specific to the authenticator.  
+
+General checks include:
+
+**NOTE:** Steps that require developer intervention to implement the status check are italicized below
+1. The requesting user has access to the authenticator status webservice
+    1. See the security section [here](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1062) for more info about this webservice 
+1. _The authenticator implements the status check_
+1. The account exists
+1. The webservice exists
+1. The authenticator is enabled in the ENV
+1. _Checks specific requirements for the new authenticator_
+
+To properly implement the status check functionality for the new authenticator, you will need to expand on italicized checks 2 and 6. 
+As described in step 2, the status of an authenticator method is checked for implementation. Therefore, you will need to add a new class in the structure of `Authentication::Authn<Type>::Authenticator` and add a status method like so:
+
+**NOTE:** It is important to keep the `Authn<Type>` pattern when defining the class structure. 
+
+For example, for OIDC, the structure would be `Authentication::AuthnOIDC::Authenticator`.
+
+```ruby
+def status(authenticator_status_input:)
+  Authentication::Authn<Type>::ValidateStatus.new.(
+    # inputs for the authenticator status check
+  )
+end
+```
+
+**NOTE:** This class is already implemented for existing authenticators and therefore, you will only to add the `status` method for this step
+
+This status method will call the ValidateStatus’s CommandClass and run the new authenticator-specific checks as described in step 6. This new 
+ValidateStatus CommandClass should be in the structure of `Authentication::Authn<Type>::ValidateStatus` and will need to be in a format similar to:
+
+**NOTE:** It is important to keep the `Authn<Type>` pattern when defining the class structure. 
+
+For example, for OIDC, the structure would be `Authentication::AuthnOIDC::ValidateStatus`.
+
+```ruby
+module Authentication
+  module Authn<Type>
+
+    Err = Errors::Authentication::Authn<Type>
+    # Possible Errors Raised:
+      
+    ValidateStatus = CommandClass.new(
+      dependencies: {
+      },
+      inputs: %i()
+    ) do
+    
+        def call
+          specific_authn_check1
+          specific_authn_check2
+          .
+          .
+        end
+    
+        private
+        
+        def specific_authn_check1 
+        end
+    
+        def specific_authn_check2
+        end
+      end
+   end
+end
+
+```
+
+**NOTE:** For an example of an Authenticator ValidateStatus CommandClass see [here](https://github.com/cyberark/conjur/blob/master/app/domain/authentication/authn_oidc/validate_status.rb)
```

Comment:
please add an example like this for the new `status` method described above: https://github.com/cyberark/conjur/blob/master/app/domain/authentication/authn_oidc/authenticator.rb#L12

Reply:
just the status method? or also `valid?`

Reply:
just the `status`. i meant that the example should come after the part where you demonstrate the `status` method.